django_socio_grpc/middlewares.py

```diff
@@ -0,0 +1,74 @@
+import asyncio
+import logging
+
+from asgiref.sync import sync_to_async
+from django import db
+from django.utils.decorators import sync_and_async_middleware
+
+from django_socio_grpc.servicer_proxy import GRPCRequestContainer
+from django_socio_grpc.settings import grpc_settings
+from django_socio_grpc.utils.utils import safe_async_response
+
+logger = logging.getLogger("django_socio_grpc")
+
+
+@sync_and_async_middleware
+def close_old_connections_middleware(get_response):
+    if asyncio.iscoroutinefunction(get_response):
+
+        async def middleware(request):
+            db.reset_queries()
+            await sync_to_async(_close_old_connections)()
+            try:
+                return await safe_async_response(get_response, request)
+            finally:
+                await sync_to_async(_close_old_connections)()
+
+    else:
+
+        def middleware(request):
+            db.reset_queries()
+            _close_old_connections()
+            try:
+                return get_response(request)
+            finally:
+                _close_old_connections()
+
+    return middleware
+
+
+@sync_and_async_middleware
+def log_requests_middleware(get_response):
```

Comment:
can you add some comment and link to grpc doc about how you make this working ? and not just the function all the part of the function